---
layout: minimal_post
title: Work Log from 2015-06-25 to the present
date: 2015-06-25 10:00:00 
published: true
---

* Friday, July 24, 2015
    * Met with Thomas and helped advise him on some issues with the NGS simulator he is writing.
    * Met with Anthony and helped him with some strategies for parsing VCF files.
    * Agreed to do a review for _Molecular Ecology_.
    
    
* Thursday, July 23, 2015
    * Anthony and I Skyped with Ying at UCLA and Kristen to discuss some of their concerns about the
    data coming out of Stacks.  They have rather remarkable Hardy-Weinberg issues. An excess of homozygotes
    that really can't be attributed to population structure... It seems that perhaps there is something
    wonky about Stacks's VCF output.  This provided a nice segue into a talk with Anthony about managing the
    VCF output from FreeBayes.
    * Spent some time installing vcftools and reading the documentation, as we are going to be getting into
    this with FreeBayes.  Good stuff...
    * Working on sib-prog-assess.
    * At night, big hours working with VCF output and familiarizing myself with workflows for handling it.
    
----
    
* Wednesday, July 22, 2015, afternoon/evening
    * Met with Ryan Waples and Garrett Kinney at UW.  Discussed pipelines for RAD data, usage of `gsi_sim`
    and the possibility of accounting for linkage disequilibrium in GSI analyses with 10K RAD loci.
    * Editing the sib-prog-assess paper to incorporate the new findings from running colony version 2.0.5.2.

* Monday through noon Wednesday, July 20-22, 2015
    * At the SISG on leave status.  
    * But did 2 hours of work on Monday night maintenance on R packages.

* Sunday, July 19, 2015
    * 4 hours working on sib-prog-assess (getting all the running time data recorded and
    presented) and writing on paper with MVB and HS.  

* Friday, July 17, 2015
    * Met with Thomas and we discussed his progress on the short read simulator script. We also
    did some research together into calling haplotypes of short reads and ANGSD and FreeBayes, and 
    VCF formats.
    * Major progress on the sib-prog-assess paper.  Incorporated the new version of Colony into all the
    main text figures and just have a few more supplemental ones to deal with.

* Thursday, July 16, 2015
    * So glad to be done with that review, getting back to other things.
    * Met with Carlos and Alicia to discuss revisions to the Mex Trout paper at PLOS One.
    * Met with Anthony and we talked about modifying our NGS Bioinformatics pipeline to
    cut Stacks out of the mix (except maybe for the de novo assembly part).
    * Reading/editing the current draft of the closekin paper for StatSci that Mark B and Hans
    S. have been writing.
    * Helped Simone with an R question...
    * Working on sib-prog-assess. I added lines to boxplot script to plot all the results from the
    new version as well.  Started trying to make the aggregated smearograms.  Bit of a disaster
    as things don't play well with `psfrag` any longer.  Grrr.... I might just annotate it with
    R directly and not worry about axes having nicely typeset mathematics.
    

---- 

* Wednesday, July 15, 2015
    * Met with Thomas and reviewed some pedfactory code with him for an hour.
    * Writing on that TPB review.  I should be finishing that this afternoon or tonight. Yep!
    Finished it this afternoon!
    * Looking over Stacks output trying to figure out how to help Diana condense things into
    a good set of candidates.
 

* Tuesday, July 14, 2015
    * From noon on...
    * Lab meeting.
    * Helping Diana crunch out some information from stacks.  It looks like an expedient way to
    do that at this point, quickly, might be to just copy and paste from the stacks view of good
    looking candidate loci and then do something like this:
    
    ```r
    con <- pipe("(echo; pbpaste; echo)")
    a <- readLines(con)
    close(con)
    
    a2 <- a[str_detect(a, "^[ACGT][ACGT][ACGT]")] # grab the named haplotypes
    ```
    And then, at that point just record the observed genotypes into a data frame in R so we can
    analyze the patterns later.  I'm a little concerned about the high number of observed
    homozygotes in the stacks output...
    * The stacks mysql data base is really wonky.  Anthony and I were talking about using Stacks
    only for the de-novo assembly, and then using bwa-mem to align the reads to the consensus 
    sequences of the stacks after first lumping the stacks that are very similar into a single
    consensus.

* Monday, July 13, 2015
    * On leave.
    
* Friday, July 10, 2015
    * 2 more hours family leave.  Kristen had some med appts and I was doing drop-off etc for Z.
    * Review for TPB and reading widely on pairwise relationship inference from dense
      genomic data.

* Thursday, July 9, 2015
    * 2 hours family leave (Kristen had dental work and I was with zoe)
    * Working on that TPB review. Good stuff.  Learning lots in reading the background papers.

----

* Wednesday, July 8, 2015
    * Agreed to do a review for MEPS.
    * Email to Garrett Kinney with help on gsi_sim.  He is in the Seeb lab.  It is  great
    that gsi_sim is finding use there.
    * Refining and optimizing the Structure trace and results plot maker.
    * Working on TPB review
    * Reviewing Thomas' python code for the amplicon sequencing SNP calling simulator project.
      we decided that using the coalescent to simulate polymorphisms on there would probably
      the the best way forward.
    
    

* Tuesday, July 7, 2015
    * Working through some stuff with SNPPIT.  I had helped out Sophie Gerber a while ago, but she
    has selfing populations.  It turns out that when SNPPIT sees selfing parents, SNPPIT finds
    it more probable that the selfing is a situation where each candidate is a full sibling of the
    offspring.  THis is because selfing is not included 
    * Lab meeting.
    * At night, working on TPB review.
    
    
* Monday, July 6, 2015
    * Read through the Big Creek paper and sent comments back to Devon and the other coauthors.
    * Worked on modifying scripts to do final plots for the sib-prog-assess.
    * Met with thomas.  Discussed his work on assessing ANGSD vs FreeBayes for calling SNPs (and
    haplotypes if possible...with FreeBayes, perhaps) from amplicon sequencing data.  Also reviewed
    some of the code he is writing for implementing the pedigree factor graph inference engine for
    a single-connected case.
    
* Friday, July 3, 2015
    * Independence day observed.
    
* Thursday, Jul 2, 2015
    * Wrapping up edits for the SISG course this year.
    * Also completed a package for visualizing trace and Q value concordance between `structure`
    runs.  Check out an explanation [here](http://rpubs.com/eriqande/scot-cats).  I made it for
    pedagogical purposes, but I think it will be a nice thing to have here in the lab for sussing
    out convergence of different structure runs.
  
----

* Wednesday, Jul 1, 2015
    * Revamping some slides on the computer practical for the SISG.
    * Reviewing devtools-based package development in R.

* Tuesday, June 30, 2015
    * Crunching out a quick R-based wrapper for `structure` for ggplotting of results and traces.
    * Discussing with Thomas about an accounting system for loops in pedigree factor graphs.

* Monday, June 29, 2015
    * Family leave.

* Fri Jun 26, 2015
    * 2 hours family leave in the morning.
    * Working on improving gsi_sim. By the afternoon I had put in and tested some new code to
    avoid underflow when you have 1,000s of loci.  That is good.  One of the folks at UW 
    (Garrett McKinney) had run into problems with 15,000 SNPs.  The changes to the code can
    be seen at [here](https://github.com/eriqande/gsi_sim/commit/280734b4d7c4bd1a89f2bd5273c3bd660b5d73e7).
    * Working on revisions of TPB paper.
    



* Thu Jun 25, 2015
    * Checked in with Anthony in morning and we talked about the GSI project, NGS, bioinformatics,         and most of all about Bayescan and Lositan and how we interpret Martha's results from 
      those programs.
    * Helping Martha extract some genotypes from particular loci that Bayescan identifies as outliers, but which manifestly are not.
    * Computer maintenance stuff.
    * Reading up on the Junction Tree algorithm, trying to understand that.  It is relevant to pedfactory calculations.
    * Looking over Thomas' results on computing the values of different messages coming out of libDAI.
    * At night looking over gsi_sim code to figure out how to fix an underflow problem when 1,000s
    of loci are used.

----  
