---
layout: minimal_post
title: Work Log from 2015-06-25 to the present
date: 2015-06-25 10:00:00 
published: true
---

* Wednesday, Mar 9, 2016
    * Edits on that Seagrant proposal with Lorenz.
    * Helped Colin Nicol with some R stuff.
    * Dealing with lot of email and software requests and questions.

* Tuesday, Mar 8, 2016
    * Catching up on email, getting back in touch with Thierry, cleaning out my box.
    * Green Sturgeon finalizing analyses. Looking at the matching pairs---nothing
      exciting there.  
    * Some editing on aquaculture grandparentage preproposal. 
    * Had wanted to start writing up some of the Bayesian hierarchical reporting unit stuff,
      but will have to start that up tomorrow.
    * Talked with Thomas about Sr.Microhap.  He is going to take a whack at implementing the
      MCMC for it.
      
* Monday, Mar 7, 2016
    * Worked with Anhtony on rockfish species ID loci.
    * All hands meeting.
    * Skyped with Brendan Wringe and Ryan in Ian Bradbury's lab.  Talking with them about
      how to avoid various types of bias when assessing the power of a selected panel of 
      SNPs for use in NewHybrids.
    * Reading about `packrat` and updating R and Rstudio on my system.
    * Finalizing the sturgeon meta-data issues.

    
* Friday, Mar 4, 2016
    * Working with Thierry on assigner and talking with him about RAD seq issues, etc.
    
* Thursday, Mar 3, 2016
    * Thierry came to visit.  Before he arrived I worked on writing a few functions to
      automate the process of downloading and compiling gsi_sim for his R package
      assigner.

---

* Wednesday, Mar 2, 2016
    * email and admin stuff
    * working with anthony and carlos on rockfish species ID from sea lion poop.
    * working with thomas on Bio::cigar stuff.
    
* Tuesday, Mar 1, 2016
    * Sturgeon fiddling.  Got everything ready to go, just need the final determination on some
      of the IDs.
    * Working with Thomas on pedfactory next steps.
    
* Monday, Feb 29, 2016
    * Some writing/formulating on Sr.Microhap
    * Incorporating the reconciled meta data into the green sturgeon analyses and I 
      finished transferring Carlos' edits over to the document Rnw file.
      
* Friday, Feb 26, 2016
    * Green sturgeon meta data wrangling.
    * Writing up the theory on sr.microhap
    * Skype conference with the WIWA isotope folks.
    
* Thursday, Feb 25, 2016
    * Finalizing changes to SNPPIT to avoid underflow.  Yay! It seems to be working!
      Merging and pushing all those changes.
    * Talked with Diana about statistical issues in parentage and rockfish.
    * Working out the mathematical details of a short-read microhaplotype caller
      that we can (I hope) trust.

---

* Wednesday, Feb 24, 2016
    * Scheming with Thomas about a haplotype inferring method from 
      short read data for our amplicon purposes.  
    * I gave the guest lecture, "Applications of graphs in Statistical 
      and Probabilistic Inference" to UCSC Math 115. Advanced       
      undergraduate course in Graph Theory, taught by Richard
      Montgomery.

* Tuesday, Feb 23, 2016
    * Putting together a guest lecture for a graph theory course in
      the math dept.

* Monday, Feb 22, 2016
    * Rockfish haplotyping woes. Discussions with Anthony and bringing Thomas on this,
      then also met with everyone on the project in the afternoon.
    * Writing a guest lecture for a math class.
    

* Friday, Feb 19, 2016
    * Bioinformatics fest with Anthony figuring out how to score haplotypes from 
      amplicon sequencing data.
    * Phone conference with Kristen and Kristina Paxton.  They are trying to add
      isotope data to the wilson's warbler genetic assignments.
    * Later on into the wee hours of the night I finished up that wiwa stuff.

---

* Wednesday, Feb 17, 2016
    * Wrapping up review of lobster erratum and sending it off.
    * Catching up on email.
    * talking with anthony about SNP selection
    * Working on a guest lecture for the Math 115: Graph Theory.  
    
    
* Tuesday, Feb 16, 2016
    * Finishing up simulations of het deficit in RAD data's effects on GSI accuracy
      prediction: not much effect...
    * Presented the Nature Rev Genet paper on RAD-like techniques for lab meeting, and 
      mostly talked about the issue of PCR duplicates and of my simulations.
    * Reviewing Louis' lobster erratum.
    

* Friday, Feb 12, 2016
    * Preparing a presentation for lab meeting whilst thinking about the homozygote excess in RAD data.
    * Doing more simulations for that.
    
* Thursday, Feb 11, 2016
    * Thinking through and doing some simulations to see if mis-scoring of heterozygotes as 
      homozygotes (as happens in RAD seq with lots of PCR duplicates) might bias predicted power
      of GSI.  

---

* Wednesday, Feb 10, 2016
    * Finished that review for _Mol Ecol Res_.
    * Crunched out some data from UCLA on their RAD PCR duplicate levels.  That is a
      serious problem if you can filter out the dupes (as you can't with ddRAD or
      single-end RAD.)
    * This came out in November on CJFAS's "Just In":  Daniel J. Hasselman, Eric C Anderson, 
      Emily E Argo, N. David Bethoney, Stephen R Gephard, David M. Post, Bradley P. Schondelmeier,
      Thomas F Schultz, Theodore V Willis, Eric P Palkovacs. 2015. Genetic stock composition of
      marine bycatch reveals disproportional impacts on depleted river herring genetic stocks.
      Canadian Journal of Fisheries and Aquatic Sciences, Published on the web 
      19 November 2015, 10.1139/cjfas-2015-0402
    * Left an hour early to take zoe to the OrthoD.

* Tuesday, Feb 9, 2016
    * 1 hours family leave. consult with orthodontist for Zoe.
    * A bunch of email with people using NewHybrids, etc.
    * Admin stuff---ITM recertification.
    * Working on review for _Mol Ecol Res_.
    * Lab meeting.
    
* Monday, Feb 8, 2015
    * Some correspondence with Louis Bernatchez.
    * Crafted some code to send Colin Wallace at DFO to help him
      crunch out gsi_sim results.
    * Tracking down sturgeon discrepancies with Carlos.
    * Working on CKMRsim.
    * Advancement celebration lunch for Thomas!
    * Working on a review for _Mol Ecol Res_.
    
    
    
* Friday, Feb 5, 2016
    * Finished that review for NPRB.
    * Talking with Anthony about the variant calls.
    * Had a call from Tim Vines to talk about Axios Review.

* Thursday, Feb 4, 2016
    * NPRB grant review.
    * lobster checkin stuff
    * video conference with Chris H.
    * Talking chrom 5 with carlos and devon.
---

* Wednesday, Feb 3
    * Working on a grant review for the NPRP.
    * Went with Thomas to his advancement, which he passed.  Yay!
    * Looking over and discussing sturgeon paper with Carlos.
    
    
* Tuesday, Feb 2
    * Helped Clemento with some VCF-handling AWK goodness.
    * Finished that review for _Heredity_.
    * Looking over some graph theory stuff.
    * Email correspondence with Ryan Waples re: loss matrices and gsi_sim.
    
    
* Monday, Feb. 1
    * Meeting with Thomas and coaching him on his advancement seminar a bit.
    * Looking over Richard Montgomery's algebraic graph theory course syllabus as I will be
      guest-lecturing on Feb 24 there.
    * Writing up the review for _Heredity_.
    

* Saturday, Jan 23 to  Sunday, Jan 31.
    * I was mostly on leave, but did work on several things during the
    time: 
        - Two hours developing CKMRsim.  I have the simulation all done and am working on 
          efficiently turning all the simulated values into alphas and betas while dealing
          with different pi's in a coherent fashion.
        - 4 hours working on a review for _Heredity_.  This review required more background
          reading than I thought it would originally.  I have figured it all out nicely now,
          though.
        - 2 hours working on final edits to the _Stat-Sci_ close-kin paper, which should receive
          final acceptance in another day or two.  

* Friday, Jan 22, 2016
    * Fixing some issues with GSI_SIM that DFO ran into.
    * Showing Thomas and Anthony how to run ms2geno.
    * Working on fixing a SNPPIT gtyp probability underflow problem.
    

* Thursday, Jan 21, 2016
    * Major CKMR coding.  Great progress!! 
    * Meeting with Anthony on the freebayes/VCF saga...

---

* Wednesday, Jan 20, 2016
    * recompiling newhybrids on linux for a user
    * CKMR.
    * Working with thomas on the number of individuals involved in matings that create cycles each
      generation.

* Tuesday, Jan 19, 2016
    * CKMR
    * Lab meeting. And afterward working with Thomas to prep his advancement.

* Friday, Jan 15, 2016
    * CKMR

* Thursday, Jan 14, 2016
    * More CKMR programming.
    * Helping Thomas with some editing on his advancement proposal.

* Wednesday, Jan 13, 2016
    * Finished describing, then implemented and tested the CKMRsim generalized 
      allele-based genotyping error model.  Now making convenience function for 
      microhaplotypes.  
    * Went to campus for Peter Moyle's seminar.
    * Helped Diana with some R programming stuff.

* Tuesday, Jan 12, 2016
    * More CKMRsim writing and formulating of the error model. Thought through how to deal
      with null alleles and other incompletely observed genotypes.  I think that can fit
      happily into the framework I have developed.
    * Lab meeting.
    * Helping Thomas debug his `pedfactory` code.  We found the problem! He was inputting 
      1-p instead of p in for the allele frequency.  He fixed that and now the program is giving
      lovely results.  
    * Talking with Devon and Carlos about potential postdoc hires.

* Monday, Jan. 11, 2016
    * CKMRsim.  I spent much of the day formulating a general genotyping error model to
      use with it, and writing about it for the paper.
    * Meeting with Thomas about his advancement proposal.
    * Working up some calculations in my sum-product-mockup to check some wonky results that
      Thomas as gotten.
    
* Friday, Jan. 8, 2016
    * CKMRsim development and coding.  I am working on creating the X_l matrices of 
      pairwise genotype probabilities, and it became quite clear after I coded it up
      in R that I really need to revamp it in Rcpp, cuz it was incredibly slow and space
      inefficient in R.  So, doing that now, and it is going well...Yep...finished it and
      it rips.  

* Thursday, Jan. 7, 2016
    * Started off with cleaning up the sturgeon bycatch maps considerably, and getting hi-resolution
      coastlines in there.  I found a page with a `multiplot` function on how to arrange ggplots
      within the grid environment.  I really need to get better with `grid`.
    * CKMRsim programming.
    * Helping Thomas with some editing.
    
---

* Wednesday
    * Finished that review for _Evol Apps_
    * Helping Thomas track down some bugs/weird behavior in pedfactory.
    * some CKMRsim stuff.
    
    
* Tuesday
    * Turning sturgeon report into a single column and sending it to the coauthors.
    * lab meeting.
    * meeting with Thomas re: pedfactory priors.  
    * night, writing more functions for Rrunstruct to help extract allele freqs from the results.
    * a bit of loony stuff with kristen.

* Monday
    * Final revisions on close kin.
    * Review for Evol Apps

* Friday, Jan 1, 2016
    * New Year's Day

* Thursday, Dec 31, 2015
    * Working on some bird stuff with kristen.  we have a ddRAD data set from Mike Sorensen
      of 152 loons from around North America and I started putting it through its paces to see
      what kind of structure there is and whether that could be recovered with a subset of 96 SNPs.
---

* Wednesday, Dec 30, 2015
    * Morning: closekin revisions got an extension till Jan 5, but I had momentum
      so finished all my revisions.
      Hans is travelling for a while now, but it should be ready to submit very soon.
    * Afternoon: editing the sturgeon thing a little.
    * Working on a review for Evol Apps.
    
* Tuesday, Dec 29, 2015
    * More madness trying to pull the closekin revisions together.  It is close.

* Monday, Dec 28, 2015
    * All green sturgeon all day.
        - Wrote code for doing the self-ID stuff via likelihood and it is amazing
        - Generated figures for the paper
        - Started writing up the results
        - Cool figure for number of successfully scored assays by chip.
        - Sent the manuscript off to Carlos.


* Thursday, Dec 24, 2015
    * Green sturgeon.  Writing up how to do the self-id using likelihood.

---

* Wednesday, Dec 23, 2015
    * Got a note from Colin Wallace at DFO saying that gsi_sim is working well for them.
    * Working with Hans and Mark on the closekin paper revisions.
    * Getting stuff done on the green sturgeon report.

* Tuesday, Dec 22, 2015
    * finishing up changes to snppit and also to gsi_sim.  The latter is now compiled up for windows and Linux and the DFO folks are using it because their amplicon sequencing data sets have overwhelmed ONCOR and cBayes.


* Monday, Dec 21, 2015
    * All day getting snppit cleaned up and compiling on Linux, and setting up a test arena for it. See all the commits for more details.
    
    
* Friday, Dec, 18, 2015
    * Gave a presentation and had a discussion with Eileen Sobeck and Will Stelle about how our research and work ties in with the goals of hathery reform.  I felt like that went well.  
    * Met with Thomas.
    
* Thursday, Dec 17, 2015
    * Helping APK with some spip questions for some simulations he is doing with Hankin.
    * Prepping for a presentation/discussion with Eileen Sobeck and Will Stelle.
    
---

* Wednesday, Dec 16, 2015
    * All day editing the close-kin paper.

* Tuesday, Dec 15, 2015
    * Added a function to count up discordant genotype pairs in the sturgeon analysis.
    * Revising close-kin paper.
    * lab meeting.
    
* Monday, Dec 14, 2015
    * Writing/revision on the close kin paper.

* Friday, Dec 11, 2015
    * Home ill.  Worked 3 hours on sturgeon stuff.

* Thursday, Dec 10, 2015
    * Reading up on mark-recapture with individual-specific genotypes for the revisions for the
      paper with Hans.
    * Finished that review for _Evolution_.

---

* Wednesday, Dec 9, 2015
    * Pretty much all day working on a review for _Evolution_ that is due very soon.
    
* Tuesday, Dec 8, 2015
    * Pretty much the whole day cranking on sturgeon fluidigm.  Done with sorting out the meta data
      for the most part, and worked today and preparing figures for the paper and running it through
      structure via the slg_pipe.
      
* Monday, Dec 7, 2015
    * Working on sturgeon stuff.  Writing the methods, trying to piece together what crandall did,
      working on describing our manual scoring system.
      

* Friday, Dec 4, 2015
    * Pretty much all day working over the sturgeon meta data and writing.
    * A fair bit at night on this too.

* Thursday, Dec 3, 2015
    * Lots of correspondence with Terry Beacham and some work to recompile snppit for windows with 
      a bug-fix.
    * Helping simone with some ADMB compilation issues.
    * Working on sturgeon.  Got more meta data from Carlos.  That will help things along nicely.
    
    
---

* Wednesday, Dec 2, 2015
    * Most of the day working on sturgeon stuff.  The organization of all the data and the various
      steps of the scoring wasn't super clear/reproducible the way Thomas had organized it, so I was
      tidying that up and then starting to work over the meta data.

* Tuesday, Dec 1, 2015
    * Final wrapping up of code and data for herring paper Dryad archive.
    * Correspondence with Terry Beacham re: SNPPIT.
    * Agreed to do a review for _Evol Apps_ and a re-review for _Bioinformatics_.
    * Lab meeting
    * Met with Thomas, worked on green sturgeon, discussed his advancement.
    
    
    
    
* Monday, Nov 30, 2015
    * Catching up on email etc.
    * Lots of talking with Anthony about the PyRAD vs Stacks results on the Marmoratus.
    * Working on sturgeon analysis and report.
    * Verifying reproducility of herring analysis package and getting ready to dump it to Dryad.
    

* Sunday, Nov 29, 2015
    * Got a little more reading on Rstudio Shiny's API.
    
* Friday, Nov 27, 2015
    * Two hours working on a review for _Evolution_.
    * On leave the rest of the day.

* Thursday, Nov 26, 2015
    * Thanksgiving.

---

* Wednesday, Nov 25, 2015
    * On leave

* Tuesday, Nov 24, 2015
    * Working on green sturgeon stuff.  There was still a bit of work to be done to get 
      Thomas' stuff as reproducible as it needs to be to proceed.  I worked with Thomas on 
      that most of the afternoon.
    * Also helping Chante get the SNP data run through the pipeline.

* Monday, Nov 23, 2015
    * Wrapping stuff up with Chante.
    * Revising my biosketch for future grants.
    * Helping Andrew Kinziger with some spip stuff.
    * With Thomas, wrapping up final bits of sturgeon analysis.
    * Call with Dan H. Getting the herring GitHub repo all ready for archiving on Dryad.
    
* Friday, Nov 20, 2015
    * Writing up a caption for the antenna detections figure, and redoing them to reflect the exact
      criteria that Dave and Steve had used in the analysis.
    * Going line by line with Thomas over his sturgeon analyses.  Looking good.
    
* Thursday, Nov 19, 2015
    * Working up that figure of the antenna detections for the big creek paper.
      Pretty fun time.  For one part of it (that I ended up not using...) I had to
      delve into the guts of the ``ggplot_build()` function output.  Good experience.
    * Talking over the big creek paper with Devon.  
    
---

* Wednesday, Nov 18, 2015
    * Agreed to do a review for Evolution.
    * Sent off a reply to MolEcol about lobster_checkin. My plan is to send a Comment.
    * Helping clemento and libby with the breeding matrix.
    * Plotting up some Big Creek stuff to better show what the raw data look like.
    * CAPS results meeting with Carlos.
    * Helping libby more.  Her version of Kingroup spits results out in a different format than
      Clemento's, so his Breeding scripts weren't quite working.

* Tuesday, Nov 17, 2015
    * Was all psyched to continue on the big creek stuff when I got a message from
      MolEcol saying that $(LB)^2$ and friends had sent their lobster erratum, pretty much 
      unchanged, to MolEcol.  So, I have to deal with that a little bit.
    * Lab meeting.
    * Got to fiddle with the antenna data from Big Creek a little bit.
      
      

* Monday, Nov 16, 2015
    * Morning spent talking over Big Creek paper issues with Devon.  I am feeling
      like it would be more satisfying to have some figures that actually tried to
      show the data and tell the story with the data, rather than highly processed
      summaries.  I also want to get my head around the actual data better.  And,
      some of the figures just need revamping, so I am going to work on that a little bit.
    * Revamping the figures of length vs genotype and also of length vs sex.
    * Looking at the mark-recapture and antenna data to figure out how to make better
      

* Friday, Nov 13, 2015
    * More righteous R wrangling to help set up our VCF filtering and viewing pipeline
      with Anthony.
    * Working over the Omy5 PNAS draft from Devon and Carlos.
    * Reading up on and trying out Hadley's new `purrr` package.

* Thursday, Nov 12, 2015
    * Amazing productive session with Anthony ripping through some VCF files and plotting the
      distribution of all the INFO fields.  That was super informative and it gave Anthony
      a starting place, code-wise, to start ripping out the FORMAT fields in the individual
      genotypes.
    * Working on CKMRsim.
    * Some email with newhybs users.
    
---   
    
* Wednesday
    * Vet's Day

* Tuesday, Nov 10, 2015
    * Finished up a message on the lobster thing and started working on CKMRsim.  yay.
    * Met with Thomas and we looked over the sturgeon stuff.
    
* Monday, Nov 9, 2015
    * Dealing with electrical storm shutdowns, and in the process saw that Dan Barshis was
      up in steerage. Talked with him and Anthony for a bit.
    * All hands meeting with Cisco.
    * Getting to the bottom of inbox.
    * Some lobster correspondence and code.  Ready to put that thing on the back burner
      for a while and get back to sturgeon and CKMRsim.


* Thursday, Nov 5, 2015
    * Cleaning up lobster things and writing to Laura and company.
    * Working with Anthony to rewrite the spawning matrix code using dplyr in R.  Big fun.
 
---   

* Wednesday, Nov 4, 2015
    * Doing some more simulations related to the lobster stuff.
    * Working with Chante on streamtrees.
    
* Tuesday, Nov 3, 2015
    * Helping out David Kulig. with some snppit stuff.
    * lab meeting.
    * some lobster saga
    * had a nice chat with Thierry about genomics and stackr

* Monday,  Nov 2, 2015
    * Out for 2.5 hours in the morning for massage work on my back.
    * Watched a practice talk by Kristen Ruegg for her lecture at the migratory ecology workshop
      in Lund, Sweden.  She is talking about her bird stuff, but she is also going to give a shout out
      on the Omy5 stuff as a neat example of a case where a large fraction of the variation in migratory
      behaviour seems to be predicted by the genotype at a single Mendelian locus.
    * Took a phone call from David Kuligowski at the NWFSC helping him to trouble shoot some SNPPIT issues. He
      had a few formatting problems with a massive 35Mb data file.
    * Finished updating my CV and research statement for reappointment in IMS.
    * Finished up some coding on the lobster issue.  I need to reply to Benestan now to let her know that
      her proposed erratum does a rather large disservice to the community in further confusing the
      issue on high-grading bias.


* Friday, Oct 30, 2015
    * Getting through my email.
    * Discussing recursive algorithms with Chante.
    * Discussing some priors and Bayesian issues with Will S.
    * Updating my CV to give to the IMS folks for appt renewal.
    * Met with Thomas.
    * Indicated my interest in a GBS workshop in Laval (but recall now that international
      travel on that could be a nightmare....)

* Thursday, Oct 29, 2015
    * Gave a guest lecture in MSCI 341 "Marine Population Genetics and Evolution" at CSUMB.
      The course is being taught by Dr. Alison Haupt.  I told the students about our use of SNPs
      for parentage based tagging in salmon, and then walked them all through some examples of 
      population assignment.

----

* Wednesday, Oct 28, 2015
    * Answering some NewHybrids questions for Jinliang Wang.
    * 40 minute phone call with Meredith Bartron and Jeff Kalie at FWS.  They are working
      on pallid and shovelnose sturgeon in the plains states and have been using newhybrids
      for some questions, and they were asking about how to analyze data with a fixed prior
      for the different hybrid categories.
    * Admin-stuff for the Hollings Scholarship.
    * Took the "accurate TA" course.
    * Working on figuring out which sturgeon would be good to grab for a second round.
    
      
* Tuesday, Oct 27, 2015
    * Working over some lobster stuff.  Laura sent a proposed erratum that I don't really agree with
      on all points (or at least I think it needs verification.)
    * Lab meeting.
    * Meeting with Thomas.  Checking out his Viz package.  Sweet.

* Monday, Oct 26, 2015
    * Traveling.  But I worked 2 hours on hierarchial priors for allele frequency for GSI.  
      Stuff that I think will be good for Ben Moran the Hollings Scholar to incorporate next summer.

* Friday Oct 23, 2015
    * Finished snps2assays
    * Talking with anthony about the pyRAD results.  Sounds pretty cool.  And we
      talked about looking at the effect of the different assembly methods on the
      hz deficit that we see often in RAD data run through `stacks`.
    * Sending the PC version of newhybrids off to Jinliang.
    
      
* Thursday, Oct 22, 2015
    * Working more on snps2assays
    * Goodbye lunch for hasselman

* Wednesday, Oct 21, 2015
    * Working a bit on snps2assays.
    * Met with Chante and talked about STREAMTREES.
    * Went to the EEB seminar of Tuljapurkar's
    
    
* Tuesday, Oct 20, 2015
    * At the lab I was busy configuring my system to build and test software on Windows.  
    I finally seem to have gotten
    it all set up.  I blogged about it
    [here](http://eriqande.github.io/2015/10/20/working-in-windows-minimally.html).
    * Met with Thomas and checked out his awesome progress debugging pedfactory and also 
      talked with him and Anthony about the results of the pyRad run that Thomas did on
      alewife.
    * Then into the wee hours of the night I worked on an R-package for converting SNP calls from
      RAD or ddRAD into SNP assay orders.  Put it up on GitHub
      [here](https://github.com/eriqande/snps2assays)
      
      
* Monday, Oct 19, 2015
    * 2 hours in the morning doing email stuff. Then...
    * 6 hours sick leave resting a messed up back.

* Weekend.
    * Couple of hours programming and dealing with email from Hollings scholars.

* Friday, Oct 16, 2015
    * Installing Virtual Box to run windows for Cross compiling.  
    * Phone call with Dilly.
    * Installing MSYS and MinGW for compiling on Windows.  Yay! I got it all
      working and was able to compile newhybrids (without the GUI) for Windows
    * NSF Rockfish meeting
    
    
* Thursday, Oct 15, 2015
    * 2 hours family leave morning to cover zoe.
    * CKMRsim.  Thinking through another variance reduction technique for the mixture
      distribution.  Not sure how that will work with the importance sampling.  
    * Working on setting up my computer to be able to compile things up for Windows
      as Jinliang is doing some interesting work that he wants a new version of
      NewHybrids for.
    * Extracting some assayable SNPs from the Swainson's RAD data that Kristen is going to
      design into new SNP-type assays.  I used R for this (whereas 3 years ago I would have used
      a bunch of Unix tools.  With `dplyr`, R has become significantly better for a lot of things!)

* Wednesday, Oct 14, 2015
    * Working on CKMRsim
    

* Tuesday, Oct 13, 2015
    * Finished up lobster-checkin'
    * Met with Chante for a while, helping with colony and thinking about networks
    * Met with Thomas to review his pedfactory work and talk about pyrad, etc. He
      wondered if it would be possible to use LDhat to try to simultaneously order 
      markers in the chrom 5 region.  Might be possible because we have so many populations
      that we could do it hierarchically.  Hmm...tough though...
    * Working on CKMR sim interface to mendel: slurping the mendel output and turning it into
      genotypes.
    
* Monday, Oct 12, 2015
    * Columbus Day holiday, but I got an hour of dealing with email done.

* Friday, Oct 9, 2015
    * Some work on sib-prog-assess after a bit of input from Jinliang Wang.
    * Handling some email that has stacked up.
    * Some correspondence with Jinliang Wang about expanding `structure`'s Q values
      into things that dealt with genotype freq categories.
    * Meeting with Thomas about priors on pedigrees.
    * Meeting with Carlos re: NGS in the lab.
    * Meeting with Chante.

* Thursday, Oct 8, 2015
    * Wrapping up my work trying to reproduce Benestan et al.'s lobster result into a 
    reproducible Rmd.

* Wednesday, Oct 7, 2015
    * 3 hours of outreach in the morning.  I accompanied a field trip of 32 3rd graders from
      Gateway School to the San Lorenzo river in Henry Cowell State Park.  I was one of four scientists
      [a hydrologist (USGS), an ornithologist (UCLA), a general biologist (Elkhorn Slough), and me] who
      met with students in groups of 8 at a time.  We discussed what a watershed is and the ecological
      connections between different parts of the watershed, the animals and plants that depend on the
      river, and the thermal tolerances of steelhead.  Then we helped the students take some 
      water temperature measurements for the San Lorenzo River Alliance.
    * Working more on that lobster issue.
    * Meeting with Carlos, Thomas, and Anthony to discuss future bioinformatics directions in the lab.
    * 2 hours sick leave for an appt. in the afternoon.

* Tuesday, Oct 6, 2015
    * Replied to a Hollings scholar from Stanford who had a question about working here next
      summer.  She might be a perfect candidate for fixing some GSI wonkiness.
    * Working on lobster-checkin'
    * Lab meeting.
    * Went to campus for a math seminar.
    
      
* Monday, Oct 5, 2015
    * Got intrigued by the Benestan et al paper on Lobster and started to reanalyze some of their
      data because they have what appears to be a strange results viz-a-viz high grade bias.

* Friday, Oct 2, 2015
    * Got together my code to interface with Mendel on CKMRsim.  Have it up and running for 
      sending files to Mendel, and still need to slurp Mendel input in.  Shouldn't be too hard
      but R is slow, so I will first do a version with sed and awk.

* Thursday, Oct 1, 2015
    * Helped Lisa Krigsman with some R stuff.
    * Getting up to speed on Mendel, since I think it may have some features that allow it 
      to work in CKMRsim better than Morgan.
    * Writing R code to interface with Morgan for CKMRsim
    * Met with Thomas.

---

* Wednesday, Sept 30, 2015
    * Signed the agreements for the TPB paper. They are going to have the proofs to me next
      week.  That was fast...
    * Some rep-res correspondence with Melissa about GitHub and the rockfish assessment template.
    * Final push on the sib-prog-assess paper.  Sent it off to Tony for review.  Yay! See big commit
      right [here](https://github.com/eriqande/sib-prog-assess/commit/2c7162b42c9e9ca0c09cb2ea31c697fe52ffdfb7)
    * Correspondence with Dieta Hanson (Andrew Hendry's student) about NewHybrids.
    
    
* Tuesday, Sept 29, 2015
    * Working on sib-prog-assess paper.
    * Went to the NMFS seminar on skates.
    * Met with Chante, helping her with some pipeline stuff.
    * Lab meeting.
    * Trying to see why snppit is failing on the DFO ion torrent data set.
    
* Monday, Sept 28, 2015
    * Most of the day rewriting the sib-prog-assess paper to incorporate the new colony version results,
      and make sure everything is there.

* Friday, Sept 25, 2015
    * 4 hours working on a scheduled message passing algorithm for cycle cataloguing in pedigree
      factor graphs.

* Thursday, Sept 24, 2015
    * Some correspondence with Mark Bravington.  The southern Bluefin tuna consortium likes his 
      CKMR stuff and it will move forward.  He is interested in organizing a workshop on modern
      genotyping technologies appropriate for CKMR, and was wondering if I would like to participate.
      I said yes.
    * Reading the documentation for MENDEL which may end up being more appropriate to CKMR-sim than
      MORGAN is.
    * Working on review for TPB.  Finished it!
    * Talking with Anthony about those BLAT to themselves results, and suggesting ways of going 
      forward on that.  Interesting!
    * Met with Chante for about an hour.  Signed a semi-annual update report for her.  Talked over Fst  stuff.

---

* Wednesday, Sept 23, 2015
    * Finished revisions on the TPB paper and sent them in.  Yee-ha! That was is on its way!
    * Finished all analyses with the new version of colony for the sib prog assess paper.  I have recoded stuff
      to generate figures and tables that incorporate the new results.  Now that all remains is some rewriting
      of the text to reflect these new findings.  
    * Working on a review for TPB.
    * Talking with Thomas about Depth First Searching for cycles.
    
* Tuesday, Sept 22, 2015
    * 2 hours sick leave for dentist appt.
    * Working on review for TPB.
    * Lab meeting.
    * Doing revisions for TPB paper.
    

* Monday, Sept 21, 2015
    * Working on a review for TPB.
    * Met with Carlos for performance meeting.
    * Left at midday to go the the dr. office about crazy pain in the right side of my throat.
      4 hrs sick leave.
    
    
* Friday,  Sept 18, 2015
    * Helped EJ Dick talk through some really cool hierarchical Bayesian stuff on stock assessment.
    * Talked with Anthony about the interesting RAD-BLAT in different species.
    * A few minutes helping Ying get some of the SWTH RAD data.
    * Helping Chante with the pipeline.
    * Talking with Thomas about loops in graphs.
    * sib-prog-assess.  Re-did the X-plots and the colony hi-confidence errors.  So close now!
    * finishing up year end achievements.

* Thursday,  Sept 17, 2015
    * Getting very close to wrapping up the sib-prog-assess project.
    * Also, got the revised reviews back on pedfactory.  It will be accepted, but I have to add a
      citation or two.
      
---

* Wedesday, Sept 16, 2015
    * Most of the day compiling my year-end achievements, and a little bit of sib-prog-assess.
    
* Tuesday, Sept 15, 2015
    * Exploring alternative ways to specify markers and alleles and chromosome positions to send things
      MORGAN from CKMR-sim.  Once again, long-format is way better than nested lists, and 
      dplyr rocks it.  
    * Lab meeting.  I presented my progress so far on CKMR-sim.

* Monday, Sept 14, 2015
    * Bangin' away at the CKMR-sim.  Edited the methods section a little bit and then
      started putting the R-package together.  Much time getting MORGAN figured out.
      
* Friday, Sept 11, 2015
    * Sat in on the Herring phone call with Eric P., Dan, Carlos, and Diana.
    * Helping Chante out with some R stuff.
    * Writing on CKRMsim paper.
    
    
* Thursday, Sept 10, 2015
    * Showed Thomas how to use my ECA_Opt3 command line parser library for the pedigree
    factor graph program he is putting together.
    * Sent the closekin paper with MVB to Simone for an internal review.
    * Working on writing the paper to go along with my CKMR-sim package.
    * Helping Chante with some R stuff.
    

---


* Wednesday, Sept 9, 2015
    * Started writing up the Close-Kin-Mark-recapture-sim paper.  Got some nice ideas on how to 
    implement a simulation to assess power for pairwise relationship inference from different
    types of markers, using importance sampling.  Gonna talk about that in lab meeting next week.
    * Helping Chante get settled.
    * Met with Thomas for a good long going-over of his pedfactory code.  He's nailed it.  He has the
    sum-product algorithm up and running and is not ready to start sampling over pedigrees.  Cool stuff!
    
* Tuesday, Sept 8, 2015
    * Summarized Mendelian incompatibilities in the Alsea data steelhead data.
    * Went to Juan Lopez's dissertation defense.
    * Lab meeting.
    * Working on pairwise relationship power predictor package.  (Thought I might try to bang that
    out so I can talk about it next week in lab meeting.)

* Monday, Sept 7, 2015
    * Labor Day.

* Friday, Sept 4, 2015
    * Talked with Anthony about Repeat Masking.
    * Met with Thomas and looked over some of his pedfactory code.
    * Analysed mike Blouin's mykiss data with SNPPIT and found that the panel of SNPs we have is
    quite sufficient for doing what they need to do.
    * Working on another review for TPB.
    
* Thursday, Sept 3, 2015
    * Finished a review for _Bioinformatics_.
    * Went to the CAPS presentation that K Koch did.
    
  
  
---


* Wednesday, Sept 2, 2015
    * Declined a chance to review an MS for Heredity (too many active reviews, currently)
    * Wrapping up one review for TPB.
    * Working on a review for _Bioinformatics_.
    * Helping Simone with getting set up to run Colony.
    * Helping Diana with a glitch in her pipeline run.
    
    
* Tuesday, Sept 1, 2015
    * Working on review for TPB.
    * Meeting with Thomas.
    * Lab meeting.

* Saturday, August 29, 2015
    * 7 hours, final push for final editing of Stat Sci paper section.  

* Thursday, August 27, 2015
    * 2 hours in the morning on Stat Sci paper stuff and Chante Davis arrival prep.

---

* Wednesday, August 26, 2015
    * 5 hours throughout the day debugging my importance sampling routine and getting great results from
    it on the power boost of likelihood methods vs exclusion for parent-offspring pairs.  
    * Working on writing the stat-sci piece.
* Tuesday, August 25, 2015
    * 3.5 hours in the evening, setting up an importance sampling routine to demonstrate just how much more
    power you get for parent pair inference using a likelihood ratio criterion, (rather than exclusion), even
    when genotyping error can be assumed to be zero.
* Monday, August 24, 2015
    * 3 hours in the evening working on MVBs comments on my stat-sci paper segment, handling email, and
    reading over an RFP for central valley stuff.
* Friday, August 21, 2015
     * Helping Anthony with some rxy breeding matrix operations that he is going to implement in R.
     * Doing a review for TPB.
     * Working with Devon to finalize a figure and some numbers for the Big Creek paper.

* Thursday, August 20, 2015
    * Finished writing on the StatSci paper and also finished inserting citations.
    * Spoke with Diana about the rockfish project.

---

* Wednesday, August 19, 2015
    * Agreed to do a program note review for _Bioinformatics_.
    * Working on the StatSci close kin paper.
    * More help for Anthony on the GSI project analyses.
    
* Tuesday, August 18, 2015
    * Writing on the close kin paper.  Coming along well, but bumping up against the deadline soon!
    * Helping Anthony sift through some of Crandall's old code for plotting GSI project results.
    
* Monday, August 17, 2015
    * Finished my revisions for TPB and resubmitted the pedfactory paper there.  Woo-hoo!
    * Writing on the StatSci section.
    * Tweaking colony outputs on Big Creek.
    * Catching up with Carlos.
    * Reading about array and bait capture technologies.
    * Talking with Thomas about NGS sim.

* Friday, August 14, 2015
    * Got one hour of writing on the StatSci paper.  Otherwise I was out
    at a funeral for my father's sister.  Hot day in the Central Valley.

* Thursday, August 13, 2015
    * Emailing with Robin Waples, Kelly Barr, Mark Bravington, Mandi Finger re: a variety of things.
    * Analyzing the new colony output for the big creek data set and sending that to devon
    * Talking with Anthony about the GT-seq stuff and FreeBayes.
    * Working on that section for the close-kin MR paper with Ha ns and Mark.
    * Helping Alicia with some BungData stuff.

----

* Wednesday, August 12, 2015
    * Got word that the flock comment has been published online.
    * Getting Dan H. some citations for the river herring GSI paper.
    * Working with Devon on the big creek figures.  Remade the genotype freqs vs length
    bin figures and the sibship clusters.  There will still be some final tweaking on those.
    We are re-running colony with both sexes polygamous. That will take a while to finish,
    and once that it has I will update the figure.  
    * Finishing writing up formal review of the the SBT proposal.  Sent it off.
    * Discussing the TPB paper revisions with Thomas.

* Tuesday, August 11, 2015
    * Finishing up the revisions on the TPB paper.
    * Accepted a request to referee a revision of a paper for TPB.
    * Lab meeting.
    * Writing up formal review of the Southern Bluefin Tuna proposal.
    
    
* Monday, August 10, 2015
    * Looking over some more stuff from Mark Bravington re: estimating N vs Ne using the
    close-kin methods
    * Met with Thomas and Anthony about projects.
    * Finished that review for Mol Ecol. 
    * Working on revising TPB paper.

* Saturday, August 8, 2015
    * 1 hour working on a review for MEPS
    * 1 hour drafting a review for Mol Ecol, longhand then
    * 2 more hours finishing up most of the writing and typing on that.  Just need to 
    let it age for a day, edit it, and then send it in.

* Thursday, August 5, 2015
    * 1 hour custodial stuff.  Cleaning up email, reading Robin Waples' responses to MVB, etc.
    
    
----

* Wednesday, August 4, 2015
    * 2 hours designing, formulating an R package for doing simulations to assess power
    for close kin identification with linkage and using ddRAD-like data.  This was inspired
    by MVB's work, but will be very useful predicting outcomes on our rockfish project.

* Tuesday, August 3, 2015
    * 2 hours writing up my comments for MVB and emailing them to him.

* Monday, August 2, 2015
    * 1 more hour on review for MEPS

* Saturday, July 31, 2015
    * 1 hour working on a review for MEPS

* Friday, July 31, 2015
    * 4 hours on plane reading over Mark's proposal to the CCSBT for close-kin mark-recapture.
    Robin Waples and I are acting as outside commentators on this.  Some interesting stuff about
    using baited ddRAD for GBS for kinship estimation.

* Thursday, July 30, 2015
    * More work on the TPB revisions
    * Finishing up modifications to NewHybrids to allow many more markers.
    * Met with Thomas and we went over solutions to the difficult of simulating 
    NGS data from really short RAD-like fragments.
    * Got some email from Mark Bravington with a review he was hoping I could do.

* Wednesday, July 29, 2015
    * Cranking on the TPB revisions.
    * Responding to some issues on NewHybrids:  see [here](https://github.com/eriqande/eca-shared/commit/1d46b81081ad5cdaf9bb30f7304c297253633514#commitcomment-12417566)
    * Spoke with Mandi Finger about a contract she just got to do RAD or Rapture on 
    Delta Smelt to try to estimate abundances.  The contract includes 2 qtr/year for 
    3 years for grad student support and she was wondering if a theoretically capable 
    grad student here would like to help with popgen analyses for a quarter or two sometime in
    the next year or two.  I told her that we did have a grad student who could do that,
    but that he was currently involved in bioinformatic work for our own lab and that I needed to
    talk to Carlos about it.
    * Talking with Anthony about the problems that the `mem-bwa` alignments to STACKS reference sequences
    seem to be having due to the paired end reads seeming to be scrambled.

* Tuesday, July 28, 2015
    * Two hours of family leave while the family was recoiling from the recent
    tragedy at the Tannery.
    * Working on revisions for the TPB pedfactory paper.
    * Met with Santa Cruz resident Arvind Kannan, currently working for Rancho Biosciences. 
    He wanted to learn how we use SNPs in fisheries management.
    * Meeting with Anthony and Thomas.  Anthony is grappling with some trimming issues possibly
    corrupting the coherence of mate pairs.  Thomas is running into problems with simulating next
    gen sequence data on small fragments.  Most of the simulators are designed to simulate shotgun
    sequencing of fragments of whole genomic DNA.  We looked into hacking the C++ code in `mason2` to
    allow the simulation of multiple reads all starting at the same place on a fragment.
    
* Monday, July 27, 2015
    * Early morning from home, incorporated my edits into Hans' _Statistical Science_ paper and pushed
    those changes up to the repo.
    * Attended staff meeting and sent the notes to Carlos.
    * Took the IT security course.
    * Helped Simone with some SNPPIT stuff.
    * Discussed Big Creek and a few other things briefly with Devon.
    * Working on the revisions for the TPB paper.

* Friday, July 24, 2015
    * Met with Thomas and helped advise him on some issues with the NGS simulator he is writing.
    * Met with Anthony and helped him with some strategies for parsing VCF files.
    * Agreed to do a review for _Molecular Ecology_.
    
    
* Thursday, July 23, 2015
    * Anthony and I Skyped with Ying at UCLA and Kristen to discuss some of their concerns about the
    data coming out of Stacks.  They have rather remarkable Hardy-Weinberg issues. An excess of homozygotes
    that really can't be attributed to population structure... It seems that perhaps there is something
    wonky about Stacks's VCF output.  This provided a nice segue into a talk with Anthony about managing the
    VCF output from FreeBayes.
    * Spent some time installing vcftools and reading the documentation, as we are going to be getting into
    this with FreeBayes.  Good stuff...
    * Working on sib-prog-assess.
    * At night, big hours working with VCF output and familiarizing myself with workflows for handling it.
    
----
    
* Wednesday, July 22, 2015, afternoon/evening
    * Met with Ryan Waples and Garrett Kinney at UW.  Discussed pipelines for RAD data, usage of `gsi_sim`
    and the possibility of accounting for linkage disequilibrium in GSI analyses with 10K RAD loci.
    * Editing the sib-prog-assess paper to incorporate the new findings from running colony version 2.0.5.2.

* Monday through noon Wednesday, July 20-22, 2015
    * At the SISG on leave status.  
    * But did 2 hours of work on Monday night maintenance on R packages.

* Sunday, July 19, 2015
    * 4 hours working on sib-prog-assess (getting all the running time data recorded and
    presented) and writing on paper with MVB and HS.  

* Friday, July 17, 2015
    * Met with Thomas and we discussed his progress on the short read simulator script. We also
    did some research together into calling haplotypes of short reads and ANGSD and FreeBayes, and 
    VCF formats.
    * Major progress on the sib-prog-assess paper.  Incorporated the new version of Colony into all the
    main text figures and just have a few more supplemental ones to deal with.

* Thursday, July 16, 2015
    * So glad to be done with that review, getting back to other things.
    * Met with Carlos and Alicia to discuss revisions to the Mex Trout paper at PLOS One.
    * Met with Anthony and we talked about modifying our NGS Bioinformatics pipeline to
    cut Stacks out of the mix (except maybe for the de novo assembly part).
    * Reading/editing the current draft of the closekin paper for StatSci that Mark B and Hans
    S. have been writing.
    * Helped Simone with an R question...
    * Working on sib-prog-assess. I added lines to boxplot script to plot all the results from the
    new version as well.  Started trying to make the aggregated smearograms.  Bit of a disaster
    as things don't play well with `psfrag` any longer.  Grrr.... I might just annotate it with
    R directly and not worry about axes having nicely typeset mathematics.
    

---- 

* Wednesday, July 15, 2015
    * Met with Thomas and reviewed some pedfactory code with him for an hour.
    * Writing on that TPB review.  I should be finishing that this afternoon or tonight. Yep!
    Finished it this afternoon!
    * Looking over Stacks output trying to figure out how to help Diana condense things into
    a good set of candidates.
 

* Tuesday, July 14, 2015
    * From noon on...
    * Lab meeting.
    * Helping Diana crunch out some information from stacks.  It looks like an expedient way to
    do that at this point, quickly, might be to just copy and paste from the stacks view of good
    looking candidate loci and then do something like this:
    
    ```r
    con <- pipe("(echo; pbpaste; echo)")
    a <- readLines(con)
    close(con)
    
    a2 <- a[str_detect(a, "^[ACGT][ACGT][ACGT]")] # grab the named haplotypes
    ```
    And then, at that point just record the observed genotypes into a data frame in R so we can
    analyze the patterns later.  I'm a little concerned about the high number of observed
    homozygotes in the stacks output...
    * The stacks mysql data base is really wonky.  Anthony and I were talking about using Stacks
    only for the de-novo assembly, and then using bwa-mem to align the reads to the consensus 
    sequences of the stacks after first lumping the stacks that are very similar into a single
    consensus.

* Monday, July 13, 2015
    * On leave.
    
* Friday, July 10, 2015
    * 2 more hours family leave.  Kristen had some med appts and I was doing drop-off etc for Z.
    * Review for TPB and reading widely on pairwise relationship inference from dense
      genomic data.

* Thursday, July 9, 2015
    * 2 hours family leave (Kristen had dental work and I was with zoe)
    * Working on that TPB review. Good stuff.  Learning lots in reading the background papers.

----

* Wednesday, July 8, 2015
    * Agreed to do a review for MEPS.
    * Email to Garrett Kinney with help on gsi_sim.  He is in the Seeb lab.  It is  great
    that gsi_sim is finding use there.
    * Refining and optimizing the Structure trace and results plot maker.
    * Working on TPB review
    * Reviewing Thomas' python code for the amplicon sequencing SNP calling simulator project.
      we decided that using the coalescent to simulate polymorphisms on there would probably
      the the best way forward.
    
    

* Tuesday, July 7, 2015
    * Working through some stuff with SNPPIT.  I had helped out Sophie Gerber a while ago, but she
    has selfing populations.  It turns out that when SNPPIT sees selfing parents, SNPPIT finds
    it more probable that the selfing is a situation where each candidate is a full sibling of the
    offspring.  THis is because selfing is not included 
    * Lab meeting.
    * At night, working on TPB review.
    
    
* Monday, July 6, 2015
    * Read through the Big Creek paper and sent comments back to Devon and the other coauthors.
    * Worked on modifying scripts to do final plots for the sib-prog-assess.
    * Met with thomas.  Discussed his work on assessing ANGSD vs FreeBayes for calling SNPs (and
    haplotypes if possible...with FreeBayes, perhaps) from amplicon sequencing data.  Also reviewed
    some of the code he is writing for implementing the pedigree factor graph inference engine for
    a single-connected case.
    
* Friday, July 3, 2015
    * Independence day observed.
    
* Thursday, Jul 2, 2015
    * Wrapping up edits for the SISG course this year.
    * Also completed a package for visualizing trace and Q value concordance between `structure`
    runs.  Check out an explanation [here](http://rpubs.com/eriqande/scot-cats).  I made it for
    pedagogical purposes, but I think it will be a nice thing to have here in the lab for sussing
    out convergence of different structure runs.
  
----

* Wednesday, Jul 1, 2015
    * Revamping some slides on the computer practical for the SISG.
    * Reviewing devtools-based package development in R.

* Tuesday, June 30, 2015
    * Crunching out a quick R-based wrapper for `structure` for ggplotting of results and traces.
    * Discussing with Thomas about an accounting system for loops in pedigree factor graphs.

* Monday, June 29, 2015
    * Family leave.

* Fri Jun 26, 2015
    * 2 hours family leave in the morning.
    * Working on improving gsi_sim. By the afternoon I had put in and tested some new code to
    avoid underflow when you have 1,000s of loci.  That is good.  One of the folks at UW 
    (Garrett McKinney) had run into problems with 15,000 SNPs.  The changes to the code can
    be seen at [here](https://github.com/eriqande/gsi_sim/commit/280734b4d7c4bd1a89f2bd5273c3bd660b5d73e7).
    * Working on revisions of TPB paper.
    



* Thu Jun 25, 2015
    * Checked in with Anthony in morning and we talked about the GSI project, NGS, bioinformatics,         and most of all about Bayescan and Lositan and how we interpret Martha's results from 
      those programs.
    * Helping Martha extract some genotypes from particular loci that Bayescan identifies as outliers, but which manifestly are not.
    * Computer maintenance stuff.
    * Reading up on the Junction Tree algorithm, trying to understand that.  It is relevant to pedfactory calculations.
    * Looking over Thomas' results on computing the values of different messages coming out of libDAI.
    * At night looking over gsi_sim code to figure out how to fix an underflow problem when 1,000s
    of loci are used.

----  
