---
layout: minimal_post
title: Work Log from 2015-06-25 to the present
date: 2015-06-25 10:00:00 
published: true
---

* Wednesday, Oct 21, 2015
    * Working a bit on snps2assays.
    * Met with Chante and talked about STREAMTREES.
    * Went to the EEB seminar of Tuljapurkar's
    
    
* Tuesday, Oct 20, 2015
    * At the lab I was busy onfiguring my system to build and test software on Windows.  
    I finally seem to have gotten
    it all set up.  I blogged about it
    [here](http://eriqande.github.io/2015/10/20/working-in-windows-minimally.html).
    * Met with Thomas and checked out his awesome progress debugging pedfactory and also 
      talked with him and Anthony about the results of the pyRad run that Thomas did on
      alewife.
    * Then into the wee hours of the night I worked on an R-package for converting SNP calls from
      RAD or ddRAD into SNP assay orders.  Put it up on GitHub
      [here](https://github.com/eriqande/snps2assays)
      
      
* Monday, Oct 19, 2015
    * 2 hours in the morning doing email stuff. Then...
    * 6 hours sick leave resting a messed up back.

* Weekend.
    * Couple of hours programming and dealing with email from Hollings scholars.

* Friday, Oct 16, 2015
    * Installing Virtual Box to run windows for Cross compiling.  
    * Phone call with Dilly.
    * Installing MSYS and MinGW for compiling on Windows.  Yay! I got it all
      working and was able to compile newhybrids (without the GUI) for Windows
    * NSF Rockfish meeting
    
    
* Thursday, Oct 15, 2015
    * 2 hours family leave morning to cover zoe.
    * CKMRsim.  Thinking through another variance reduction technique for the mixture
      distribution.  Not sure how that will work with the importance sampling.  
    * Working on setting up my computer to be able to compile things up for Windows
      as Jinliang is doing some interesting work that he wants a new version of
      NewHybrids for.
    * Extracting some assayable SNPs from the Swainson's RAD data that Kristen is going to
      design into new SNP-type assays.  I used R for this (whereas 3 years ago I would have used
      a bunch of Unix tools.  With `dplyr`, R has become significantly better for a lot of things!)

* Wednesday, Oct 14, 2015
    * Working on CKMRsim
    

* Tuesday, Oct 13, 2015
    * Finished up lobster-checkin'
    * Met with Chante for a while, helping with colony and thinking about networks
    * Met with Thomas to review his pedfactory work and talk about pyrad, etc. He
      wondered if it would be possible to use LDhat to try to simultaneously order 
      markers in the chrom 5 region.  Might be possible because we have so many populations
      that we could do it hierarchically.  Hmm...tough though...
    * Working on CKMR sim interface to mendel: slurping the mendel output and turning it into
      genotypes.
    
* Monday, Oct 12, 2015
    * Columbus Day holiday, but I got an hour of dealing with email done.

* Friday, Oct 9, 2015
    * Some work on sib-prog-assess after a bit of input from Jinliang Wang.
    * Handling some email that has stacked up.
    * Some correspondence with Jinliang Wang about expanding `structure`'s Q values
      into things that dealt with genotype freq categories.
    * Meeting with Thomas about priors on pedigrees.
    * Meeting with Carlos re: NGS in the lab.
    * Meeting with Chante.

* Thursday, Oct 8, 2015
    * Wrapping up my work trying to reproduce Benestan et al.'s lobster result into a 
    reproducible Rmd.

* Wednesday, Oct 7, 2015
    * 3 hours of outreach in the morning.  I accompanied a field trip of 32 3rd graders from
      Gateway School to the San Lorenzo river in Henry Cowell State Park.  I was one of four scientists
      [a hydrologist (USGS), an ornithologist (UCLA), a general biologist (Elkhorn Slough), and me] who
      met with students in groups of 8 at a time.  We discussed what a watershed is and the ecological
      connections between different parts of the watershed, the animals and plants that depend on the
      river, and the thermal tolerances of steelhead.  Then we helped the students take some 
      water temperature measurements for the San Lorenzo River Alliance.
    * Working more on that lobster issue.
    * Meeting with Carlos, Thomas, and Anthony to discuss future bioinformatics directions in the lab.
    * 2 hours sick leave for an appt. in the afternoon.

* Tuesday, Oct 6, 2015
    * Replied to a Hollings scholar from Stanford who had a question about working here next
      summer.  She might be a perfect candidate for fixing some GSI wonkiness.
    * Working on lobster-checkin'
    * Lab meeting.
    * Went to campus for a math seminar.
    
      
* Monday, Oct 5, 2015
    * Got intrigued by the Benestan et al paper on Lobster and started to reanalyze some of their
      data because they have what appears to be a strange results viz-a-viz high grade bias.

* Friday, Oct 2, 2015
    * Got together my code to interface with Mendel on CKMRsim.  Have it up and running for 
      sending files to Mendel, and still need to slurp Mendel input in.  Shouldn't be too hard
      but R is slow, so I will first do a version with sed and awk.

* Thursday, Oct 1, 2015
    * Helped Lisa Krigsman with some R stuff.
    * Getting up to speed on Mendel, since I think it may have some features that allow it 
      to work in CKMRsim better than Morgan.
    * Writing R code to interface with Morgan for CKMRsim
    * Met with Thomas.

---

* Wednesday, Sept 30, 2015
    * Signed the agreements for the TPB paper. They are going to have the proofs to me next
      week.  That was fast...
    * Some rep-res correspondence with Melissa about GitHub and the rockfish assessment template.
    * Final push on the sib-prog-assess paper.  Sent it off to Tony for review.  Yay! See big commit
      right [here](https://github.com/eriqande/sib-prog-assess/commit/2c7162b42c9e9ca0c09cb2ea31c697fe52ffdfb7)
    * Correspondence with Dieta Hanson (Andrew Hendry's student) about NewHybrids.
    
    
* Tuesday, Sept 29, 2015
    * Working on sib-prog-assess paper.
    * Went to the NMFS seminar on skates.
    * Met with Chante, helping her with some pipeline stuff.
    * Lab meeting.
    * Trying to see why snppit is failing on the DFO ion torrent data set.
    
* Monday, Sept 28, 2015
    * Most of the day rewriting the sib-prog-assess paper to incorporate the new colony version results,
      and make sure everything is there.

* Friday, Sept 25, 2015
    * 4 hours working on a scheduled message passing algorithm for cycle cataloguing in pedigree
      factor graphs.

* Thursday, Sept 24, 2015
    * Some correspondence with Mark Bravington.  The southern Bluefin tuna consortium likes his 
      CKMR stuff and it will move forward.  He is interested in organizing a workshop on modern
      genotyping technologies appropriate for CKMR, and was wondering if I would like to participate.
      I said yes.
    * Reading the documentation for MENDEL which may end up being more appropriate to CKMR-sim than
      MORGAN is.
    * Working on review for TPB.  Finished it!
    * Talking with Anthony about those BLAT to themselves results, and suggesting ways of going 
      forward on that.  Interesting!
    * Met with Chante for about an hour.  Signed a semi-annual update report for her.  Talked over Fst  stuff.

---

* Wednesday, Sept 23, 2015
    * Finished revisions on the TPB paper and sent them in.  Yee-ha! That was is on its way!
    * Finished all analyses with the new version of colony for the sib prog assess paper.  I have recoded stuff
      to generate figures and tables that incorporate the new results.  Now that all remains is some rewriting
      of the text to reflect these new findings.  
    * Working on a review for TPB.
    * Talking with Thomas about Depth First Searching for cycles.
    
* Tuesday, Sept 22, 2015
    * 2 hours sick leave for dentist appt.
    * Working on review for TPB.
    * Lab meeting.
    * Doing revisions for TPB paper.
    

* Monday, Sept 21, 2015
    * Working on a review for TPB.
    * Met with Carlos for performance meeting.
    * Left at midday to go the the dr. office about crazy pain in the right side of my throat.
      4 hrs sick leave.
    
    
* Friday,  Sept 18, 2015
    * Helped EJ Dick talk through some really cool hierarchical Bayesian stuff on stock assessment.
    * Talked with Anthony about the interesting RAD-BLAT in different species.
    * A few minutes helping Ying get some of the SWTH RAD data.
    * Helping Chante with the pipeline.
    * Talking with Thomas about loops in graphs.
    * sib-prog-assess.  Re-did the X-plots and the colony hi-confidence errors.  So close now!
    * finishing up year end achievements.

* Thursday,  Sept 17, 2015
    * Getting very close to wrapping up the sib-prog-assess project.
    * Also, got the revised reviews back on pedfactory.  It will be accepted, but I have to add a
      citation or two.
      
---

* Wedesday, Sept 16, 2015
    * Most of the day compiling my year-end achievements, and a little bit of sib-prog-assess.
    
* Tuesday, Sept 15, 2015
    * Exploring alternative ways to specify markers and alleles and chromosome positions to send things
      MORGAN from CKMR-sim.  Once again, long-format is way better than nested lists, and 
      dplyr rocks it.  
    * Lab meeting.  I presented my progress so far on CKMR-sim.

* Monday, Sept 14, 2015
    * Bangin' away at the CKMR-sim.  Edited the methods section a little bit and then
      started putting the R-package together.  Much time getting MORGAN figured out.
      
* Friday, Sept 11, 2015
    * Sat in on the Herring phone call with Eric P., Dan, Carlos, and Diana.
    * Helping Chante out with some R stuff.
    * Writing on CKRMsim paper.
    
    
* Thursday, Sept 10, 2015
    * Showed Thomas how to use my ECA_Opt3 command line parser library for the pedigree
    factor graph program he is putting together.
    * Sent the closekin paper with MVB to Simone for an internal review.
    * Working on writing the paper to go along with my CKMR-sim package.
    * Helping Chante with some R stuff.
    

---


* Wednesday, Sept 9, 2015
    * Started writing up the Close-Kin-Mark-recapture-sim paper.  Got some nice ideas on how to 
    implement a simulation to assess power for pairwise relationship inference from different
    types of markers, using importance sampling.  Gonna talk about that in lab meeting next week.
    * Helping Chante get settled.
    * Met with Thomas for a good long going-over of his pedfactory code.  He's nailed it.  He has the
    sum-product algorithm up and running and is not ready to start sampling over pedigrees.  Cool stuff!
    
* Tuesday, Sept 8, 2015
    * Summarized Mendelian incompatibilities in the Alsea data steelhead data.
    * Went to Juan Lopez's dissertation defense.
    * Lab meeting.
    * Working on pairwise relationship power predictor package.  (Thought I might try to bang that
    out so I can talk about it next week in lab meeting.)

* Monday, Sept 7, 2015
    * Labor Day.

* Friday, Sept 4, 2015
    * Talked with Anthony about Repeat Masking.
    * Met with Thomas and looked over some of his pedfactory code.
    * Analysed mike Blouin's mykiss data with SNPPIT and found that the panel of SNPs we have is
    quite sufficient for doing what they need to do.
    * Working on another review for TPB.
    
* Thursday, Sept 3, 2015
    * Finished a review for _Bioinformatics_.
    * Went to the CAPS presentation that K Koch did.
    
  
  
---


* Wednesday, Sept 2, 2015
    * Declined a chance to review an MS for Heredity (too many active reviews, currently)
    * Wrapping up one review for TPB.
    * Working on a review for _Bioinformatics_.
    * Helping Simone with getting set up to run Colony.
    * Helping Diana with a glitch in her pipeline run.
    
    
* Tuesday, Sept 1, 2015
    * Working on review for TPB.
    * Meeting with Thomas.
    * Lab meeting.

* Saturday, August 29, 2015
    * 7 hours, final push for final editing of Stat Sci paper section.  

* Thursday, August 27, 2015
    * 2 hours in the morning on Stat Sci paper stuff and Chante Davis arrival prep.

---

* Wednesday, August 26, 2015
    * 5 hours throughout the day debugging my importance sampling routine and getting great results from
    it on the power boost of likelihood methods vs exclusion for parent-offspring pairs.  
    * Working on writing the stat-sci piece.
* Tuesday, August 25, 2015
    * 3.5 hours in the evening, setting up an importance sampling routine to demonstrate just how much more
    power you get for parent pair inference using a likelihood ratio criterion, (rather than exclusion), even
    when genotyping error can be assumed to be zero.
* Monday, August 24, 2015
    * 3 hours in the evening working on MVBs comments on my stat-sci paper segment, handling email, and
    reading over an RFP for central valley stuff.
* Friday, August 21, 2015
     * Helping Anthony with some rxy breeding matrix operations that he is going to implement in R.
     * Doing a review for TPB.
     * Working with Devon to finalize a figure and some numbers for the Big Creek paper.

* Thursday, August 20, 2015
    * Finished writing on the StatSci paper and also finished inserting citations.
    * Spoke with Diana about the rockfish project.

---

* Wednesday, August 19, 2015
    * Agreed to do a program note review for _Bioinformatics_.
    * Working on the StatSci close kin paper.
    * More help for Anthony on the GSI project analyses.
    
* Tuesday, August 18, 2015
    * Writing on the close kin paper.  Coming along well, but bumping up against the deadline soon!
    * Helping Anthony sift through some of Crandall's old code for plotting GSI project results.
    
* Monday, August 17, 2015
    * Finished my revisions for TPB and resubmitted the pedfactory paper there.  Woo-hoo!
    * Writing on the StatSci section.
    * Tweaking colony outputs on Big Creek.
    * Catching up with Carlos.
    * Reading about array and bait capture technologies.
    * Talking with Thomas about NGS sim.

* Friday, August 14, 2015
    * Got one hour of writing on the StatSci paper.  Otherwise I was out
    at a funeral for my father's sister.  Hot day in the Central Valley.

* Thursday, August 13, 2015
    * Emailing with Robin Waples, Kelly Barr, Mark Bravington, Mandi Finger re: a variety of things.
    * Analyzing the new colony output for the big creek data set and sending that to devon
    * Talking with Anthony about the GT-seq stuff and FreeBayes.
    * Working on that section for the close-kin MR paper with Ha ns and Mark.
    * Helping Alicia with some BungData stuff.

----

* Wednesday, August 12, 2015
    * Got word that the flock comment has been published online.
    * Getting Dan H. some citations for the river herring GSI paper.
    * Working with Devon on the big creek figures.  Remade the genotype freqs vs length
    bin figures and the sibship clusters.  There will still be some final tweaking on those.
    We are re-running colony with both sexes polygamous. That will take a while to finish,
    and once that it has I will update the figure.  
    * Finishing writing up formal review of the the SBT proposal.  Sent it off.
    * Discussing the TPB paper revisions with Thomas.

* Tuesday, August 11, 2015
    * Finishing up the revisions on the TPB paper.
    * Accepted a request to referee a revision of a paper for TPB.
    * Lab meeting.
    * Writing up formal review of the Southern Bluefin Tuna proposal.
    
    
* Monday, August 10, 2015
    * Looking over some more stuff from Mark Bravington re: estimating N vs Ne using the
    close-kin methods
    * Met with Thomas and Anthony about projects.
    * Finished that review for Mol Ecol. 
    * Working on revising TPB paper.

* Saturday, August 8, 2015
    * 1 hour working on a review for MEPS
    * 1 hour drafting a review for Mol Ecol, longhand then
    * 2 more hours finishing up most of the writing and typing on that.  Just need to 
    let it age for a day, edit it, and then send it in.

* Thursday, August 5, 2015
    * 1 hour custodial stuff.  Cleaning up email, reading Robin Waples' responses to MVB, etc.
    
    
----

* Wednesday, August 4, 2015
    * 2 hours designing, formulating an R package for doing simulations to assess power
    for close kin identification with linkage and using ddRAD-like data.  This was inspired
    by MVB's work, but will be very useful predicting outcomes on our rockfish project.

* Tuesday, August 3, 2015
    * 2 hours writing up my comments for MVB and emailing them to him.

* Monday, August 2, 2015
    * 1 more hour on review for MEPS

* Saturday, July 31, 2015
    * 1 hour working on a review for MEPS

* Friday, July 31, 2015
    * 4 hours on plane reading over Mark's proposal to the CCSBT for close-kin mark-recapture.
    Robin Waples and I are acting as outside commentators on this.  Some interesting stuff about
    using baited ddRAD for GBS for kinship estimation.

* Thursday, July 30, 2015
    * More work on the TPB revisions
    * Finishing up modifications to NewHybrids to allow many more markers.
    * Met with Thomas and we went over solutions to the difficult of simulating 
    NGS data from really short RAD-like fragments.
    * Got some email from Mark Bravington with a review he was hoping I could do.

* Wednesday, July 29, 2015
    * Cranking on the TPB revisions.
    * Responding to some issues on NewHybrids:  see [here](https://github.com/eriqande/eca-shared/commit/1d46b81081ad5cdaf9bb30f7304c297253633514#commitcomment-12417566)
    * Spoke with Mandi Finger about a contract she just got to do RAD or Rapture on 
    Delta Smelt to try to estimate abundances.  The contract includes 2 qtr/year for 
    3 years for grad student support and she was wondering if a theoretically capable 
    grad student here would like to help with popgen analyses for a quarter or two sometime in
    the next year or two.  I told her that we did have a grad student who could do that,
    but that he was currently involved in bioinformatic work for our own lab and that I needed to
    talk to Carlos about it.
    * Talking with Anthony about the problems that the `mem-bwa` alignments to STACKS reference sequences
    seem to be having due to the paired end reads seeming to be scrambled.

* Tuesday, July 28, 2015
    * Two hours of family leave while the family was recoiling from the recent
    tragedy at the Tannery.
    * Working on revisions for the TPB pedfactory paper.
    * Met with Santa Cruz resident Arvind Kannan, currently working for Rancho Biosciences. 
    He wanted to learn how we use SNPs in fisheries management.
    * Meeting with Anthony and Thomas.  Anthony is grappling with some trimming issues possibly
    corrupting the coherence of mate pairs.  Thomas is running into problems with simulating next
    gen sequence data on small fragments.  Most of the simulators are designed to simulate shotgun
    sequencing of fragments of whole genomic DNA.  We looked into hacking the C++ code in `mason2` to
    allow the simulation of multiple reads all starting at the same place on a fragment.
    
* Monday, July 27, 2015
    * Early morning from home, incorporated my edits into Hans' _Statistical Science_ paper and pushed
    those changes up to the repo.
    * Attended staff meeting and sent the notes to Carlos.
    * Took the IT security course.
    * Helped Simone with some SNPPIT stuff.
    * Discussed Big Creek and a few other things briefly with Devon.
    * Working on the revisions for the TPB paper.

* Friday, July 24, 2015
    * Met with Thomas and helped advise him on some issues with the NGS simulator he is writing.
    * Met with Anthony and helped him with some strategies for parsing VCF files.
    * Agreed to do a review for _Molecular Ecology_.
    
    
* Thursday, July 23, 2015
    * Anthony and I Skyped with Ying at UCLA and Kristen to discuss some of their concerns about the
    data coming out of Stacks.  They have rather remarkable Hardy-Weinberg issues. An excess of homozygotes
    that really can't be attributed to population structure... It seems that perhaps there is something
    wonky about Stacks's VCF output.  This provided a nice segue into a talk with Anthony about managing the
    VCF output from FreeBayes.
    * Spent some time installing vcftools and reading the documentation, as we are going to be getting into
    this with FreeBayes.  Good stuff...
    * Working on sib-prog-assess.
    * At night, big hours working with VCF output and familiarizing myself with workflows for handling it.
    
----
    
* Wednesday, July 22, 2015, afternoon/evening
    * Met with Ryan Waples and Garrett Kinney at UW.  Discussed pipelines for RAD data, usage of `gsi_sim`
    and the possibility of accounting for linkage disequilibrium in GSI analyses with 10K RAD loci.
    * Editing the sib-prog-assess paper to incorporate the new findings from running colony version 2.0.5.2.

* Monday through noon Wednesday, July 20-22, 2015
    * At the SISG on leave status.  
    * But did 2 hours of work on Monday night maintenance on R packages.

* Sunday, July 19, 2015
    * 4 hours working on sib-prog-assess (getting all the running time data recorded and
    presented) and writing on paper with MVB and HS.  

* Friday, July 17, 2015
    * Met with Thomas and we discussed his progress on the short read simulator script. We also
    did some research together into calling haplotypes of short reads and ANGSD and FreeBayes, and 
    VCF formats.
    * Major progress on the sib-prog-assess paper.  Incorporated the new version of Colony into all the
    main text figures and just have a few more supplemental ones to deal with.

* Thursday, July 16, 2015
    * So glad to be done with that review, getting back to other things.
    * Met with Carlos and Alicia to discuss revisions to the Mex Trout paper at PLOS One.
    * Met with Anthony and we talked about modifying our NGS Bioinformatics pipeline to
    cut Stacks out of the mix (except maybe for the de novo assembly part).
    * Reading/editing the current draft of the closekin paper for StatSci that Mark B and Hans
    S. have been writing.
    * Helped Simone with an R question...
    * Working on sib-prog-assess. I added lines to boxplot script to plot all the results from the
    new version as well.  Started trying to make the aggregated smearograms.  Bit of a disaster
    as things don't play well with `psfrag` any longer.  Grrr.... I might just annotate it with
    R directly and not worry about axes having nicely typeset mathematics.
    

---- 

* Wednesday, July 15, 2015
    * Met with Thomas and reviewed some pedfactory code with him for an hour.
    * Writing on that TPB review.  I should be finishing that this afternoon or tonight. Yep!
    Finished it this afternoon!
    * Looking over Stacks output trying to figure out how to help Diana condense things into
    a good set of candidates.
 

* Tuesday, July 14, 2015
    * From noon on...
    * Lab meeting.
    * Helping Diana crunch out some information from stacks.  It looks like an expedient way to
    do that at this point, quickly, might be to just copy and paste from the stacks view of good
    looking candidate loci and then do something like this:
    
    ```r
    con <- pipe("(echo; pbpaste; echo)")
    a <- readLines(con)
    close(con)
    
    a2 <- a[str_detect(a, "^[ACGT][ACGT][ACGT]")] # grab the named haplotypes
    ```
    And then, at that point just record the observed genotypes into a data frame in R so we can
    analyze the patterns later.  I'm a little concerned about the high number of observed
    homozygotes in the stacks output...
    * The stacks mysql data base is really wonky.  Anthony and I were talking about using Stacks
    only for the de-novo assembly, and then using bwa-mem to align the reads to the consensus 
    sequences of the stacks after first lumping the stacks that are very similar into a single
    consensus.

* Monday, July 13, 2015
    * On leave.
    
* Friday, July 10, 2015
    * 2 more hours family leave.  Kristen had some med appts and I was doing drop-off etc for Z.
    * Review for TPB and reading widely on pairwise relationship inference from dense
      genomic data.

* Thursday, July 9, 2015
    * 2 hours family leave (Kristen had dental work and I was with zoe)
    * Working on that TPB review. Good stuff.  Learning lots in reading the background papers.

----

* Wednesday, July 8, 2015
    * Agreed to do a review for MEPS.
    * Email to Garrett Kinney with help on gsi_sim.  He is in the Seeb lab.  It is  great
    that gsi_sim is finding use there.
    * Refining and optimizing the Structure trace and results plot maker.
    * Working on TPB review
    * Reviewing Thomas' python code for the amplicon sequencing SNP calling simulator project.
      we decided that using the coalescent to simulate polymorphisms on there would probably
      the the best way forward.
    
    

* Tuesday, July 7, 2015
    * Working through some stuff with SNPPIT.  I had helped out Sophie Gerber a while ago, but she
    has selfing populations.  It turns out that when SNPPIT sees selfing parents, SNPPIT finds
    it more probable that the selfing is a situation where each candidate is a full sibling of the
    offspring.  THis is because selfing is not included 
    * Lab meeting.
    * At night, working on TPB review.
    
    
* Monday, July 6, 2015
    * Read through the Big Creek paper and sent comments back to Devon and the other coauthors.
    * Worked on modifying scripts to do final plots for the sib-prog-assess.
    * Met with thomas.  Discussed his work on assessing ANGSD vs FreeBayes for calling SNPs (and
    haplotypes if possible...with FreeBayes, perhaps) from amplicon sequencing data.  Also reviewed
    some of the code he is writing for implementing the pedigree factor graph inference engine for
    a single-connected case.
    
* Friday, July 3, 2015
    * Independence day observed.
    
* Thursday, Jul 2, 2015
    * Wrapping up edits for the SISG course this year.
    * Also completed a package for visualizing trace and Q value concordance between `structure`
    runs.  Check out an explanation [here](http://rpubs.com/eriqande/scot-cats).  I made it for
    pedagogical purposes, but I think it will be a nice thing to have here in the lab for sussing
    out convergence of different structure runs.
  
----

* Wednesday, Jul 1, 2015
    * Revamping some slides on the computer practical for the SISG.
    * Reviewing devtools-based package development in R.

* Tuesday, June 30, 2015
    * Crunching out a quick R-based wrapper for `structure` for ggplotting of results and traces.
    * Discussing with Thomas about an accounting system for loops in pedigree factor graphs.

* Monday, June 29, 2015
    * Family leave.

* Fri Jun 26, 2015
    * 2 hours family leave in the morning.
    * Working on improving gsi_sim. By the afternoon I had put in and tested some new code to
    avoid underflow when you have 1,000s of loci.  That is good.  One of the folks at UW 
    (Garrett McKinney) had run into problems with 15,000 SNPs.  The changes to the code can
    be seen at [here](https://github.com/eriqande/gsi_sim/commit/280734b4d7c4bd1a89f2bd5273c3bd660b5d73e7).
    * Working on revisions of TPB paper.
    



* Thu Jun 25, 2015
    * Checked in with Anthony in morning and we talked about the GSI project, NGS, bioinformatics,         and most of all about Bayescan and Lositan and how we interpret Martha's results from 
      those programs.
    * Helping Martha extract some genotypes from particular loci that Bayescan identifies as outliers, but which manifestly are not.
    * Computer maintenance stuff.
    * Reading up on the Junction Tree algorithm, trying to understand that.  It is relevant to pedfactory calculations.
    * Looking over Thomas' results on computing the values of different messages coming out of libDAI.
    * At night looking over gsi_sim code to figure out how to fix an underflow problem when 1,000s
    of loci are used.

----  
